pkg/roachpb/data.go

```diff
@@ -346,7 +363,20 @@ func (v *Value) ShallowClone() *Value {
 
 // IsPresent returns true if the value is present (existent and not a tombstone).
 func (v *Value) IsPresent() bool {
-	return v != nil && len(v.RawBytes) != 0
+	if v == nil || len(v.RawBytes) == 0 {
+		return false
+	}
+	// TODO(ssd): This is a bit awkward because this is the right thing to
+	// do for production callers trying to determine if this value is a
+	// tombstone. But, many tests shove random strings into RawBytes, and in
+	// then case we'll hit this case if the 5th character of that string
+	// happens to be `e` (ascii 101). There aren't _that_ many callers to
+	// IsPresent(). We may just need to audit them all.
```

Comment:
On the larger PR, @tbg says:

> Can you address this TODO before merging?

Reply:
I can see how far we can get. Until I fully do this: https://github.com/cockroachdb/cockroach/issues/131664 it'll be possible for it to regress I think.